### PR TITLE
API changes

### LINKED ISSUE #201: Improve overall API
#### Summary
This is an issue collecting the API improvements that came up during the implementation of the new phoenix-structure into rusk (see https://github.com/dusk-network/rusk/issues/1809).

The changes are:
**CORE**
- Add function for computing `value_commitment`
- Add function for computing `transparent_value_commitment`
- Turn the value-commitment into an `JubJubAffine` point
- Expose `NOTE_ENCRYPTION_SIZE`
- Remove `sync_address` from `Note::transparent` parameter
- Make `alloc` feature a `default` feature

**CIRCUITS**
- Add `RecipientParameter::new()` constructor

### PR DESCRIPTION
**Pull request by moCello**
Resolves #201

### CODE REVIEW

File: circuits/src/recipient.rs [hunk 1]
```diff
@@ -61,6 +59,84 @@ impl Default for RecipientParameters {
     }
 }
 
+impl RecipientParameters {
+    /// Create the recipient parameter
+    pub fn new(
+        rng: &mut (impl RngCore + CryptoRng),
```
Comment (ureeves) [inline circuits/src/recipient.rs:62]: This is the recommended way of taking an rng by `rand`:
```suggestion
        rng: &mut (impl RngCore + CryptoRng + ?Sized),
```

File: core/Cargo.toml [hunk 2]
```diff
@@ -28,7 +28,7 @@ rand = "0.8"
 rkyv = { version = "0.7", default-features = false, features = ["size_32"] }
 
 [features]
-default = []    # "alloc" is suggested as default feature but would be breaking change
```
Comment (ureeves) [inline core/Cargo.toml:31]: Yup, we're breaking anyway, so might as well.

**Review by ureeves — APPROVED** [review]
LGTM. One small thing at your discretion

File: core/src/lib.rs [hunk 3]
```diff
@@ -18,17 +19,39 @@ mod note;
...
 
 #[cfg(feature = "alloc")]
 /// Transaction Skeleton used by the phoenix transaction model
 pub use transaction::TxSkeleton;
+
+use dusk_jubjub::{
+    JubJubAffine, JubJubScalar, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED,
+};
+
+/// Use the pederson commitment scheme to compute a transparent value
```
Comment (xevisalle) [inline core/src/lib.rs]: Pedersen

File: core/src/lib.rs [hunk 4]
```diff
@@ -18,17 +19,39 @@ mod note;
...
+/// commitment.
+pub fn transparent_value_commitment(value: u64) -> JubJubAffine {
+    JubJubAffine::from(GENERATOR_EXTENDED * JubJubScalar::from(value))
+}
+
+/// Use the pederson commitment scheme to compute a value commitment using a
```
Comment (xevisalle) [inline core/src/lib.rs]: Same

**Review by xevisalle — COMMENTED** [review]
just the two typos!

File: circuits/src/transaction.rs [hunk 5]
```diff
@@ -141,14 +143,10 @@ struct WitnessTxOutputNote {
 
 impl TxOutputNote {
     /// Crate a new `TxOutputNote`.
```
Comment (xevisalle) [inline circuits/src/transaction.rs]:
```suggestion
    /// Create a new `TxOutputNote`.
```

File: circuits/src/recipient.rs (hunk 1 above)
Comment (moCello) [reply circuits/src/recipient.rs:62]: Seems to introduce nasty trait bounds that I don't feel like solving now

**Review by xevisalle — APPROVED** [review]

**Pull request merged**